### PR TITLE
10039 Support for Virtual Hub default route table

### PR DESCRIPTION
**Pull request by luigibk**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: azurerm/internal/services/network/virtual_hub_resource.go [hunk 1]
```diff
@@ -192,11 +259,26 @@ func resourceVirtualHubCreateUpdate(d *schema.ResourceData, meta interface{}) er
 	}
 	d.SetId(*resp.ID)
 
+	routeTableParams := network.HubRouteTable{
+		Name:                    utils.String(defaultRouteTable),
+		HubRouteTableProperties: expandDefaultRouteTable(d.Get("defaultRouteTable").(*schema.Set).List()),
+	}
+
+	routeTableFuture, err := routeTableClient.CreateOrUpdate(ctx, resourceGroup, name, defaultRouteTable, routeTableParams)
```
Comment (timja) [inline azurerm/internal/services/network/virtual_hub_resource.go:267]: I assume this works if you don't set any default route table config?

File: azurerm/internal/services/network/virtual_hub_resource.go [hunk 2]
```diff
@@ -100,13 +101,79 @@ func resourceVirtualHub() *schema.Resource {
 				},
 			},
 
+			"defaultRouteTable": {
```
Comment (timja) [inline azurerm/internal/services/network/virtual_hub_resource.go:104]: all resources should be snake_case as far as I'm aware
```suggestion
			"default_route_table": {
```

File: azurerm/internal/services/network/virtual_hub_resource.go [hunk 3]
```diff
@@ -100,13 +101,79 @@ func resourceVirtualHub() *schema.Resource {
...
+											Type:         schema.TypeString,
+											ValidateFunc: validation.StringIsNotEmpty,
+										},
+									},
+
+									"destinations_type": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"CIDR",
```
Comment (timja) [inline azurerm/internal/services/network/virtual_hub_resource.go:142]: is there no enum value for this available?

File: azurerm/internal/services/network/virtual_hub_resource.go [hunk 4]
```diff
@@ -100,13 +101,79 @@ func resourceVirtualHub() *schema.Resource {
...
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: azure.ValidateResourceID,
+									},
+
+									"next_hop_type": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "ResourceId",
+										ValidateFunc: validation.StringInSlice([]string{
```
Comment (timja) [inline azurerm/internal/services/network/virtual_hub_resource.go:158]: I assume this is a copy from the other wan route table? although seems odd you can set this if there's only one value available

File: azurerm/internal/services/network/virtual_hub_resource.go (hunk 1 above)
Comment (timja) [reply azurerm/internal/services/network/virtual_hub_resource.go:267]: should be added as an acceptance test? https://github.com/hmcts/terraform-provider-azurerm/blob/master/azurerm/internal/services/network/virtual_hub_resource_test.go

File: azurerm/internal/services/network/virtual_hub_resource.go (hunk 4 above)
Comment (TomKimber) [reply azurerm/internal/services/network/virtual_hub_resource.go:157]: The Microsoft documentation seems to indicate they will add to this. It says something like "only ResourceId for now".

**Pull request merged**